src/main/java/org/gridsuite/modification/server/NetworkModificationController.java

```diff
@@ -167,4 +174,33 @@ public ResponseEntity<Void> stopBuild(@Parameter(description = "Build receiver")
         networkModificationService.stopBuildRequest(receiver);
         return ResponseEntity.ok().build();
     }
+
+    @GetMapping(value = "/network-modifications/line/catalog", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get a line catalog")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "The line catalog is returned"),
+        @ApiResponse(responseCode = "204", description = "The line catalaog is empty") })
+    public ResponseEntity<List<LineType>> getLineCatalog(@Parameter(description = "type") @RequestParam(name = "kind", required = false, defaultValue = "UNDEFINED") LineKind kind) {
```

Comment:
UNDEFINED is not a type
Maybe separate the endpooint to get `all` catalog lines and a endpoint to get only catalog lines for a type

Reply:
made it in the same enpoint with an optional `kind` RequestParam